buildkite/resource_pipeline_team.go

```diff
@@ -0,0 +1,250 @@
+package buildkite
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Khan/genqlient/graphql"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	resource_schema "github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type pipelineTeamResourceModel struct {
+	Id          types.String `tfsdk:"id"`
+	Uuid        types.String `tfsdk:"uuid"`
+	PipelineId  types.String `tfsdk:"pipeline_id"`
+	TeamId      types.String `tfsdk:"team_id"`
+	AccessLevel types.String `tfsdk:"access_level"`
+}
+
+type pipelineTeamResource struct {
+	client *Client
+}
+
+func newPipelineTeamResource() resource.Resource {
+	return &pipelineTeamResource{}
+}
+
+func (pipelineTeamResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_pipeline_team"
+}
+
+func (tp *pipelineTeamResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	tp.client = req.ProviderData.(*Client)
+}
+
+func (pipelineTeamResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = resource_schema.Schema{
+		MarkdownDescription: "A team pipeline resource sets a team's access for the pipeline.",
+		Attributes: map[string]resource_schema.Attribute{
+			"id": resource_schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"uuid": resource_schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"team_id": resource_schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The GraphQL ID of the team.",
+			},
+			"pipeline_id": resource_schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The GraphQL ID of the pipeline.",
+			},
+			"access_level": resource_schema.StringAttribute{
+				Required:            true,
```

Comment:
I notice the graphql mutation to create a team pipeline has `omitempty: true` for the access level. I assume thats because a team has a default member access level so its not necessary. But this marks it as required. So I think either the omitempty is wrong or this should be optional?

Reply:
I have removed the `omitempty: true` in the graphql